docs/metrics.md

```diff
@@ -0,0 +1,422 @@
+
+
+##### `consoleErrors`
+
+A count of the errors logged to the browser console. Number.
+
+
+##### `amountCached`
+
+The proportion of requests returning `304` status codes. Number, between `0.0` and `1.0`.
+
+
+##### `engagement`
+
+A normalized score tracking a user's engagement to the page; specifics are to be determined, but this will likely be similar to the [Chromium Site Engagement](https://www.chromium.org/developers/design-documents/site-engagement) metric. Number, between `0.0` and `100.0`.
+
+
+#### Session data
+
+Measurements for the _session_ to which the current page belongs. Numbers aggregated across multiple page loads are reported as an object containing the count, sum, and decile partitions, e.g.
+
+```json
+{
+    "num": 125,
+    "total": 151304,
+    "deciles": [224, 536, 766, 972, 1297, 1457, 1731, 2076, 2353]
+}
+```
+
+If there are less than 10 items in the session, the raw data is reported in lieu of `deciles`. These fields are denoted below as _aggregated_.
+
+
+##### `sessionSize`
+
+A count of unique page visits in a session. Number.
+
+
+##### `sessionLength`
+
+Milliseconds elapsed in a session. Number.
+
+
+##### `sessionTimerContentLoaded`
+
+Aggregated number of milliseconds between the start of the page load and the `DOMContentLoaded` event.
+
+
+##### `sessionTimerWindowLoad`
+
+Aggregated number of milliseconds between the start of the page load and the `load` event on `window`. 
+
+
+##### `sessionTimerFirstPaint`
+
+Aggregated number of milliseconds between the start of the page load and the first non-blank paint
+
+
+##### `sessionTimerFirstInteraction`
+
+Aggregated number of milliseconds between the start of the page load and the first of the following events:
+
+* `click`
+* `touch`
+* `scroll`
+* `keypress`
+
+Will include a count of `null` values for pages in which none of these events took place.
+
+
+##### `sessionRequests`
+
+An object containing information about requests spawned by the page. Each member key indicates the type of request:
+
+* `html`: HTML files
+* `css`: Cascading style sheets
+* `js`: JavaScript
+* `xhr`: files retrieved by `XMLHttpRequest`
+* `fonts`: non-system fonts
+* `images`: image files
+* `media`: audio and video files
+* `flash`: Flash embeds
+* `ws`: WebSocket connections
+* `other`: anything not covered in the above categories
+* `all`: all requests summed.
+
+These are categorized in the same way as the [Firefox Developer Tools Network Monitor(https://developer.mozilla.org/docs/Tools/Network_Monitor).
+
+Each value of the object contains three data points, each of which are aggregated across pages in the session:
+
+* `num`: the number of requests of that type
+* `size`: the summed filesize of requests of that type, in bytes
+* `time`: the time it took to load those filesge. Each member ke, in ty indicates the type of request:
+
+
+##### `sessionEngagement`
+
+Aggregate engagement for each page in the session.
+
+
+### Example ping
+
+```json
```

Comment:
Nit: change code fence language to "js" so it better handles the comments and `...` below.